### DISCUSSION TITLE
Why no & operation?

### DISCUSSION THREAD

**Discussion by riyueniao** [scala/scala-parser-combinators#450] [Q&A]
Why there isn't an operation like the following code.
```
// for same input, `p & q` succeeds if `p` succeeds and `q` succeeds. 
def & [U >: T](q: => Parser[U]): Parser[U] = ???
```

**Comment by peteraldous** [comment]
I'm not sure, but it isn't clear to me which result should be returned. The type signature suggests that `q`'s result would be returned, but that seems like an arbitrary choice that would surprise some users.
